org.eclipse.scanning.api/src/org/eclipse/scanning/api/CountableScannable.java

```diff
@@ -0,0 +1,87 @@
+/*-
+ *******************************************************************************
+ * Copyright (c) 2011, 2017 Diamond Light Source Ltd.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Matthew Gerring - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package org.eclipse.scanning.api;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.scanning.api.device.IScannableDeviceService;
+import org.eclipse.scanning.api.event.core.IPublisher;
+import org.eclipse.scanning.api.scan.event.Location;
+
+/**
+ * 
+ * A scannable with the ability to record method counts.
+ * 
+ * This is primarily useful for testing and debugging it
+ * means that a dependency on Mockhito is not required 
+ * in order to count method calls.
+ * 
+ * @author Matthew Gerring
+ *
+ */
+public abstract class CountableScannable<T> extends AbstractScannable<T> {
+
+
+	private Map<String, Integer> counts = new HashMap<>();
+
+	public CountableScannable() {
+		super();
+	}
+
+	public CountableScannable(IPublisher<Location> publisher, IScannableDeviceService sservice) {
+		super(publisher, sservice);
+	}
+
+	public CountableScannable(IPublisher<Location> publisher) {
+		super(publisher);
+	}
+
+	public CountableScannable(IScannableDeviceService sservice) {
+		super(sservice);
+	}
+	
+	protected void count(StackTraceElement[] ste) {
+		String methodName = getMethodName(ste);
+		Integer count = counts.get(methodName);
+		if (count==null) count = 0;
+		count = count+1;
+		counts.put(methodName, count);
+	}
+	
+	public int getCount(String method) {
+		if (!counts.containsKey(method)) return 0;
+		return counts.get(method);
+	}
+	
+	protected static final String getMethodName ( StackTraceElement ste[] ) {  

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move the array designator from the variable to the type. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1197)